mri/operators/gradient/base.py

```diff
@@ -0,0 +1,67 @@
+# #############################################################################
+#  pySAP - Copyright (C) CEA, 2017 - 2018                                     #
+#  Distributed under the terms of the CeCILL-B license,                       #
+#  as published by the CEA-CNRS-INRIA. Refer to the LICENSE file or to        #
+#  http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html for details.   #
+# #############################################################################
+
+# Internal Library import
+from .utils import check_lipschitz_cst
+
+# Third party import
+import numpy as np
+from modopt.math.matrix import PowerMethod
+from modopt.opt.gradient import GradBasic
+
+
+class GradBaseMRI(GradBasic):
+    """ Base Gradient class for all gradient operators
+        Implements the gradient of equation:
```

Comment:
It's not the gradient of the equation but the gradient of the function with respect to `x`.